google/oauth2/credentials.py

```diff
@@ -106,10 +121,12 @@ def with_scopes(self, scopes):
 
     @_helpers.copy_docstring(credentials.Credentials)
     def refresh(self, request):
-        access_token, refresh_token, expiry, _ = _client.refresh_grant(
-            request, self._token_uri, self._refresh_token, self._client_id,
-            self._client_secret)
+        access_token, refresh_token, expiry, grant_response = (
+            _client.refresh_grant(
+                request, self._token_uri, self._refresh_token, self._client_id,
+                self._client_secret))
 
         self.token = access_token
         self.expiry = expiry
         self._refresh_token = refresh_token
+        self._id_token = grant_response.get('id_token', self._id_token)
```

Comment:
After thinking a bit, these are the only scenarios:

1. The credentials are constructed with a refresh token but no id token. The credentials have the openid scope. Refresh is called, the id token is populated.
2. The credentials are constructed with a refresh token and an id token. The credentials have the openid scope. Refresh is called, the id token is updated.
3. The credentials are constructed with a refresh token and an id token. The credentials don't have the openid scope (where did that id token come from, then?). Refresh is called, the id token is preserved.

Case 3 is so weird that I'm tempted to *not* preserve the id token. So I'm gonna switch it.